### PR TITLE
Add systemd compatibility

### PR DESCRIPTION
**Pull request by alanjjenkins**
Hey Lelutin,

This is for https://github.com/lelutin/puppet-fail2ban/issues/22

This PR makes logpath optional so that this module can be used with the systemd backend as fail2ban expressly states that logpath should not be set when using the systemd backend.

I have tested this and it works but please let me know if you would like me to amend anything.

Thanks,

Alan

### CODE REVIEW

**Comment by lelutin** [comment]
hey thanks a bunch for you work :)

the TL;DR is that I'd like to make two amendments to your commits if you're ok with it, and there's also a tiny bit of manifest that needs to be added if you're up to it (otherwise I can handle it)

I see some style changes in the manifests that I would either split out into a separate commit or drop out.. namely:

 * removal of the empty line at the beginning of the class in init.pp makes the style different than the other files, so I'd simply drop this change (I can amend that during the merger, so no need to change it!)
 * changing the arrows for ordering classes in init.pp -> ah! I can see by looking at puppetlab's best practices page (style guide) that I'm following their antipattern -- "you learn something new every day" I guess :)  -- if you're ok with this, I would keep your change but simply split it out into its own commit. again there's no need to change things since I can rearrange this during the merger.

splitting out $valid_backends into a variable and reusing it for formatting the error message: I like that change 👍  thanks!

now last detail, your change in the template file is good! however I'd tend to add some validation in init.pp to avoid conflicting parameter values. so in this case, we'd have to verify that $logpath is different from false if & only if the backend is not set to systemd and the inverse would also be true.

if this last point is a bother, I can easily add it on top.. I just want to give you the chance to add the verification so that you get full credit for the code ;)

**Comment by lelutin** [comment]
Hi there Alan,

Thanks a whole bunch for your work :) All's looking good.

I've added a check for parameters to cover the error case where a user would set backend "systemd" and set a logpath, which would have been output in the config file without error.
